google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/RemoveEntireSitelinkCampaignExtensionSetting.java

```diff
@@ -0,0 +1,192 @@
+                        .build())
+            .collect(Collectors.toList());
+
+    // Creates a list of operations that contains both the campaign extension setting and the feed
+    // item operations.
+    List<MutateOperation> allOperations = new ArrayList();
+    allOperations.add(
+        MutateOperation.newBuilder()
+            .setCampaignExtensionSettingOperation(extensionSettingOperation)
+            .build());
+    allOperations.addAll(extensionFeedItemOperations);
+
+    // Connects to the API.
+    try (GoogleAdsServiceClient client =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Issues the request.
+      MutateGoogleAdsResponse response = client.mutate(String.valueOf(customerId), allOperations);
+
+      // Prints the result of removing the campaign extension setting.
+      System.out.printf(
+          "Removed a campaign extension setting with resource name: '%s'.%n",
+          response
+              .getMutateOperationResponses(0)
+              .getCampaignExtensionSettingResult()
+              .getResourceName());
+
+      // Prints the result of removing the extension feed items.
+      for (MutateOperationResponse mutateResponse :
+          response
+              .getMutateOperationResponsesList()
+              .subList(1, response.getMutateOperationResponsesList().size())) {
+        System.out.printf(
+            "Removed an extension feed item with resource name: '%s'.%n",
+            mutateResponse.getExtensionFeedItemResult().getResourceName());
+      }
+    }
+  }
+
+  /** Retrieves all sitelink extension feed items for a customer + campaign combination. */
+  private List<String> getSitelinkExtensionFeedItems(
+      GoogleAdsClient googleAdsClient, long customerId, long campaignId) {
+    // Defines a query to retrieve the sitelink extension feed items.
+    String query =
+        String.format(
+            "SELECT campaign_extension_setting.campaign, "
+                + "  campaign_extension_setting.extension_type, "
+                + "  campaign_extension_setting.extension_feed_items "
+                + "FROM "
+                + "  campaign_extension_setting "
+                + "WHERE "
+                + "  campaign_extension_setting.campaign = '%s' "
+                + "  AND campaign_extension_setting.extension_type = SITELINK",
+            ResourceNames.campaign(customerId, campaignId));
+    // Connects to the API.
+    try (GoogleAdsServiceClient client =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Issues the search request.
+      SearchPagedResponse response = client.search(String.valueOf(customerId), query);
+
+      // Constructs the result (a list of resource names matching the query).
+      List<String> result = new ArrayList();
+      response
+          .iterateAll()
+          .forEach(row -> result.add(row.getCampaignExtensionSetting().getResourceName()));
```

Comment:
This adds a campaign extension setting resource name, not the extension feed item resource name.